modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/config/AppSchemaDataAccessConfigurator.java

```diff
@@ -121,6 +121,9 @@ public class AppSchemaDataAccessConfigurator {
 
     public static final String PROPERTY_REPLACE_OR_UNION = "app-schema.orUnionReplace";
 
+    /** Whether the mapping is for an include. */
+    private final Boolean isInclude;
```

Comment:
There were a lot of tests that have calls to the method so to avoid a lot of rewrites I overloaded the method with this as an argument.  The latest push switched it to have an Optional wrapper to make it more explicit.  Does that work?

Reply:
Switched to plain boolean primitive type with default of false.